File: content/problems/1352-product-of-the-last-k-numbers/solutions/prefix-product-since-zero/Solution.kt
class ProductOfNumbers {
    private val prefix=mutableListOf(1);
    fun add(num:Int){
        if(num==0){
            prefix.clear();
            prefix.add(1)
        }else prefix.add(prefix.last()*num)
    };
    fun getProduct(k:Int):Int=if(k>=prefix.size)0 else prefix.last()/prefix[prefix.lastIndex-k]
}

File: content/problems/1352-product-of-the-last-k-numbers/solutions/prefix-product-since-zero/solution.cpp
class ProductOfNumbers {
    vector<int> prefix{
        1
    };
    public:void add(int num){
        if(!num)prefix={
            1
        };
        else prefix.push_back(prefix.back()*num);
    }
    int getProduct(int k){
        return k>=(int)prefix.size()?0:prefix.back()/prefix[prefix.size()-1-k];
    }
};
